test/e2e/framework/prometheus.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	routev1 "github.com/openshift/api/route/v1"
+)
+
+func (f *Framework) MakePrometheusWithWebTLSRemoteReceive(name, tlsSecretName string) *monitoringv1.Prometheus {
+	p := f.MakeBasicPrometheus(name, 1, []string{tlsSecretName})
+	f.AddRemoteReceiveWithWebTLSToPrometheus(p, tlsSecretName)
+	return p
+}
+
+func (f *Framework) MakeBasicPrometheus(name string, replicas int32, secrets []string) *monitoringv1.Prometheus {
+	return &monitoringv1.Prometheus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   f.Ns,
+			Annotations: map[string]string{},
+			Labels: map[string]string{
+				E2eTestLabelName: E2eTestLabelValue,
+			},
+		},
+		Spec: monitoringv1.PrometheusSpec{
+			Replicas:                        &replicas,
+			ServiceMonitorSelector:          nil,
+			ServiceMonitorNamespaceSelector: nil,
+			PodMonitorSelector:              nil,
+			PodMonitorNamespaceSelector:     nil,
```

Comment:
In fact I realize that the selectors would be nil by default so no need to set them explicitly.

```suggestion
```

Reply:
I was wondering about that, though failed to check myself.